test/qunit/unit-es5/test-component-scheduler.js

```diff
@@ -66,6 +66,37 @@ test('life cycle logic for component', function () {
     cc.js.unregisterClass(MyComponent);
 });
 
+test('component should call start before render when its node is actived', function() {
+    var nodes = createNodes({
+        comp: cc.component,
+        child: {
+            comp: CallbackTester,
+        },
+    });
+    nodes.child.active = false;
+    nodes.comp.update = function () {
+        nodes.child.active = true;
```

Comment:
nodes.child.comp.notExpect(CallbackTester.start, "start should be called asynced");
nodes.child.active = true;
nodes.child.comp.expect(CallbackTester.start, "should start in this frame");